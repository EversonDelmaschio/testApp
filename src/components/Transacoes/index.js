import React from 'react';
import {FlatList, StyleSheet, Text, View} from 'react-native';
import Transacao from '../Transacao';
import PropTypes from 'prop-types';

const Transacoes = ({transacoes}) => {
  const renderItem = ({item}) => {
    const {id, transacao, valor, data} = item;
    return (
      <Transacao
        key={`${id}-${transacao}`}
        tipo={transacao}
        valor={valor}
        data={data}
      />
    );
  };
  return (
    <View testID="transacoes" style={styles.container}>
      <Text style={styles.label}>Historico transações</Text>
      {transacoes && (
        <FlatList
          testID="transacoes"
          data={transacoes}
          contentContainerStyle={{flexGrow: 1, paddingBottom: 20}}
          renderItem={renderItem}
          ItemSeparatorComponent={() => <View style={{height: 10}} />}
          keyExtractor={item => item.id}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 15,
    paddingBottom: 20,
    flex: 1,
  },
  label: {
    fontSize: 16,
    paddingVertical: 10,
  },
  valor: {
    fontWeight: '600',
  },
});

Transacoes.defaultProp = {
  transacoes: [
    {
      transacao: '',
      valor: '0',
      data: '',
    },
  ],
};

Transacoes.propTypes = {
  transacoes: PropTypes.arrayOf(
    PropTypes.shape({
      tipo: PropTypes.string,
      valor: PropTypes.string,
      data: PropTypes.string,
    }),
  ),
};

export default Transacoes;
