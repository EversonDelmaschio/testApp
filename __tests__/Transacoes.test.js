import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import Transacoes from '../src/components/Transacoes';

describe('Component de Transações do component principal', function () {
  it('Deve renderizar uma lista de transaçoes', function () {
    const transacoes = [
      {
        data: '08/09/2022',
        valor: '20.00',
        transacao: 'saque',
      },
    ];
    const root = render(<Transacoes transacoes={transacoes} />);
    expect(root.getByText('saque')).toBeTruthy();
  });

  it('Deve renderizar igual ao snapshot', function () {
    const transacoes = [
      {
        data: '08/09/2022',
        tipo: 'saque',
        valor: '20.00',
      },
    ];
    const {toJSON} = render(<Transacoes transacoes={transacoes} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
