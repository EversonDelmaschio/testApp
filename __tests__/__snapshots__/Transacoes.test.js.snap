// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component de Transações do component principal Deve renderizar igual ao snapshot 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 15,
    }
  }
  testID="transacoes"
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(62,160,63,0.3)",
        "marginTop": 10,
        "width": "90%",
      }
    }
    testID="transacao-1"
  >
    <Text>
      08/09/2022
    </Text>
    <Text>
      
    </Text>
    <Text>
      R$ 
      20.00
    </Text>
  </View>
</View>
`;
